src/main/java/com/amazonaws/neptune/auth/NeptuneApacheHttpSigV4Signer.java

```diff
@@ -59,16 +60,59 @@ public class NeptuneApacheHttpSigV4Signer extends NeptuneSigV4SignerBase<HttpUri
      * Create a V4 Signer for Apache Commons HTTP requests.
      *
      * @param regionName             name of the region for which the request is signed
-     * @param awsCredentialsProvider the provider offering access to the credentials used for signing the request
+     * @param v1AwsCredentialProvider the provider offering access to the credentials used for signing the request
      * @throws NeptuneSigV4SignerException in case initialization fails
      */
-    public NeptuneApacheHttpSigV4Signer(
-            final String regionName, final AwsCredentialsProvider awsCredentialsProvider)
+    public NeptuneApacheHttpSigV4Signer(final String regionName,
+                                        final AWSCredentialsProvider v1AwsCredentialProvider)
             throws NeptuneSigV4SignerException {
 
+        super(regionName, v1AwsCredentialProvider);
+    }
+
+    /**
+     * Create a V4 Signer for Apache Commons HTTP requests.
+     *
+     * @param regionName             name of the region for which the request is signed
+     * @param v1AwsCredentialProvider the provider offering access to the credentials used for signing the request
+     * @param serviceName            name of the service name used to sign the requests. Defaults to neptune-db
+     * @throws NeptuneSigV4SignerException in case initialization fails
+     */
+    public NeptuneApacheHttpSigV4Signer(final String regionName,
+                                        final AWSCredentialsProvider v1AwsCredentialProvider,
+                                        final String serviceName) throws NeptuneSigV4SignerException {
```

Comment:
Should this be an enum?

Reply:
Leaving the field as free form string to avoid new changes/PRs to support new service names when added